### DISCUSSION TITLE
Intercepting API

### DISCUSSION THREAD

**Discussion by yamadaOliva** [pocketbase/pocketbase#5271] [Q&A] [answered]
I am currently facing a situation where I am using the CRUD API provided by Pocketbase. I want to check if the data already exists in Table A or meets certain conditions in Table B, and if so, return an error response. Should I use API rules for this, and can onRecordBeforeCreateRequest() handle this?
`onRecordBeforeCreateRequest((e) => {
  console.log(Object.keys(e.httpContext));
  const userId = e.record.get("requestFrom");
  const familyId = e.record.get("requestTo");
  try {
    console.log(e.httpContext);
    const result = arrayOf(
      new DynamicModel({
        id: "",
      })
    );
    $app
      .dao()
      .db()
      .select("id")
      .from("family_request")
      .where(
        $dbx.exp("requestFrom = {:id}", { id: userId }),
        $dbx.exp("requestTo = {:id}", { id: familyId })
      )
      .andWhere($dbx.exp("sys_setStatus = 'pending'"))
      .andWhere($dbx.exp("request_status = 'open'"))
      .all(result);
    if (result.length > 0) {
      throw new Error("You have already sent a request to this family");
    }
  } catch (error) {}
}, "family_request");`

**Comment by Irvingouj** [comment]
to make it readable for other readers (by ChatGPT):
```js
onRecordBeforeCreateRequest((e) => {
    // Log the keys from the httpContext to understand the available data
    console.log(Object.keys(e.httpContext));

    // Retrieve userId and familyId from the record
    const userId = e.record.get("requestFrom");
    const familyId = e.record.get("requestTo");

    try {
        // Log the full httpContext for debugging
        console.log(e.httpContext);

        // Prepare an empty array to hold the query results
        const result = [];

        // Execute the query to check existing conditions
        $app.dao().db().select("id")
            .from("family_request")
            .where($dbx.exp("requestFrom = {:id}", { id: userId }),
                   $dbx.exp("requestTo = {:id}", { id: familyId }))
            .andWhere($dbx.exp("sys_setStatus = 'pending'"))
            .andWhere($dbx.exp("request_status = 'open'"))
            .all(result);

        // If the result is not empty, throw an error to prevent the creation
        if (result.length > 0) {
            throw new Error("You have already sent a request to this family");
        }
    } catch (error) {
        // Optionally handle the error or log it
        console.error("Error processing request: ", error);
    }
}, "family_request");

```

**Reply by yamadaOliva** [reply]
thank you !!

**Reply by Irvingouj** [reply, marked as answer]
I think It is better to use onRecordBeforeCreateRequest in your usecase.

**Reply by yamadaOliva** [reply]
I tried successfully. Thanks. By the way Can you explain that case, i can not get anything. 
```javascript
onRecordBeforeCreateRequest((e) => {
  const userId = e.record.get("user");
  const familyId = e.record.get("belongTo");
  console.log(userId, familyId);
  const result = arrayOf(
    new DynamicModel({
      id: "",
    })
  );
  $app
    .dao()
    .db()
    .select("id")
    .from("family_access")
    .where(
      $dbx.and(
        $dbx.exp("belongTo = {:id}", { id: familyId }),
        $dbx.exp("user = {:id}", { id: userId })
      )
    )
    .all(result);
  try {
    $app
      .dao()
      .db()
      .select("id")
      .from("family_access")
      .where($dbx.exp("belongTo = {:id}", { id: familyId }))
      .andWhere($dbx.exp("user = {:id}", { id: userId }))
      .all(result);
  } catch (error) {
    console.log(error);
  }

  console.log(result.length);
  if (result.length > 0) {
    console.log(result[0].id);
    throw new ForbiddenError("Bạn đã là thành viên", "");
  }
}, "family_access");

```

**Reply by Irvingouj** [reply]
I'm not sure I follow. To my knowledge, you can't do multiple where clause with dynamic parameters at this time.
